### PR TITLE
[LoRA] make set_adapters() method more robust.

### PR DESCRIPTION
**Pull request by sayakpaul**
# What does this PR do?

First caught in https://github.com/huggingface/diffusers/issues/9525#issuecomment-2375371315 by @asomoza. But then it turned out to be https://github.com/huggingface/diffusers/issues/9525#issuecomment-2375988214. 

This PR:

1. Makes our `set_adapters()` method more robust by catching errors on adapter names that are wrong (and adds a test for it). 
2. Makes our existing test on `set_adapters()` more robust. 

@cshowley do you wanna give this a try?

### CODE REVIEW

**Comment by sayakpaul** [comment]
Failing tests are completely unrelated.

File: tests/lora/utils.py [hunk 1]
```diff
@@ -960,6 +972,38 @@ def test_simple_inference_with_text_denoiser_multi_adapter(self):
...
+        self.assertTrue(check_if_lora_correctly_set(denoiser), "Lora not correctly set in denoiser.")
+
+        if self.has_two_text_encoders or self.has_three_text_encoders:
+            if "text_encoder_2" in self.pipeline_class._lora_loadable_modules:
+                pipe.text_encoder_2.add_adapter(text_lora_config, "adapter-1")
+                self.assertTrue(
+                    check_if_lora_correctly_set(pipe.text_encoder_2), "Lora not correctly set in text encoder 2"
+                )
+
+        with self.assertRaises(ValueError) as err_context:
```
[the remark this answers is not in the document]
Comment (sayakpaul) [reply tests/lora/utils.py:998]: We just prefer to do a bit more explicitly across the codebase (which is what I have followed here).

File: src/diffusers/loaders/lora_base.py [hunk 2]
```diff
@@ -532,6 +532,13 @@ def set_adapters(
             )
 
         list_adapters = self.get_list_adapters()  # eg {"unet": ["adapter1", "adapter2"], "text_encoder": ["adapter2"]}
+        current_adapter_names = {adapter for _, adapter_list in list_adapters.items() for adapter in adapter_list}
```
[the remark this answers is not in the document]
Comment (sayakpaul) [reply src/diffusers/loaders/lora_base.py]: Does b0dc60db1 work?

**Review by BenjaminBossan — APPROVED** [review]
Thanks for the quick adjustments.

File: src/diffusers/loaders/lora_base.py [hunk 3]
```diff
@@ -532,13 +532,19 @@ def set_adapters(
             )
 
         list_adapters = self.get_list_adapters()  # eg {"unet": ["adapter1", "adapter2"], "text_encoder": ["adapter2"]}
-        all_adapters = {
-            adapter for adapters in list_adapters.values() for adapter in adapters
-        }  # eg ["adapter1", "adapter2"]
+        # eg ["adapter1", "adapter2"]
+        all_adapters = {adapter for adapters in list_adapters.values() for adapter in adapters}
+        missing_adapters = set(adapter_names) - all_adapters
+        if missing_adapters:
```
Comment (yiyixuxu) [inline src/diffusers/loaders/lora_base.py:538]:
```suggestion
        if len(missing_adapters) > 0:
```

**Review by yiyixuxu — APPROVED** [review]
thanks!

**Pull request merged**